### PR TITLE
Add a README for config manager authors

### PR DESCRIPTION
**Pull request by petr-muller**
This document attempts to describe the contract/conventions for all
config managers, so individual cards do not need to repeat the contract.

Not sure if this is the right place, I'll gladly move it somewhere else if needed.

[DPTP-2440](https://issues.redhat.com/browse/DPTP-2440)

### CODE REVIEW

**Review by alvaroaleman — APPROVED** [review]
/hold
Looks good modulo the configmap details. Maybe also mention it is a configmap because there might be non-public information in it?

File: cmd/branchingconfigmanagers/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+Any new config manager should be deployed to run in a periodic Prow job which mounts a ConfigMap `TBD` that contains the
+YAML file with [OCP lifecycle data](#product-lifecycle-data). The Prowjob should first run the Config Manager, then if
+it ended with zero exit code, it should use the `prcreator` tool to commit the changes and submit a PR to propagate the
+changes.
+
+TODO: What period
+
+### ConfigMap with OCP Lifecycle Data
+
+TODO: What ConfigMap
```
Comment (petr-muller) [inline cmd/branchingconfigmanagers/README.md:102]: @stevekuznetsov do we know this information already?

**Comment by petr-muller** [comment]
@alvaroaleman thanks, I added a item to the `Config Managers Must` section about the confidentiality. I'm not sure if the details about ConfigMap placement is finalized already, pinged Steve about it.

File: cmd/branchingconfigmanagers/README.md (hunk 1 above)
Comment (stevekuznetsov) [reply cmd/branchingconfigmanagers/README.md:102]: No, this is TODO

Comment (petr-muller) [reply cmd/branchingconfigmanagers/README.md:102]: I'll add a note about why this is a TODO and when it needs to be filled in, then I think we can merge, WDYT?. It's useful for CM authors even without discussing how to deploy.

**Pull request merged**